alicloud/resource_alicloud_cs_kubernetes_autoscaler.go

```diff
@@ -0,0 +1,552 @@
+		// modify the default one
+		modifyScalingConfigurationRequest := ess.CreateModifyScalingConfigurationRequest()
+		modifyScalingConfigurationResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			modifyScalingConfigurationRequest.RegionId = client.RegionId
+			modifyScalingConfigurationRequest.UserData = userData
+			modifyScalingConfigurationRequest.Tags = createScalingGroupTags(labels, taints)
+			modifyScalingConfigurationRequest.ScalingConfigurationId = defaultConfiguration.ScalingConfigurationId
+			return essClient.ModifyScalingConfiguration(modifyScalingConfigurationRequest)
+		})
+
+		if err != nil {
+			return WrapErrorf(err, "failed to invoke ModifyScalingConfiguration,because of %v", err)
+		}
+
+		addDebug("ModifyScalingConfiguration", modifyScalingConfigurationResponse, modifyScalingConfigurationRequest)
+	}
+	return nil
+}
+
+func GetScalingGroupSizeRange(client *connectivity.AliyunClient, groupId string) (min, max int, err error) {
+	describeScalingGroupRequest := ess.CreateDescribeScalingGroupsRequest()
+	describeScalingGroupRequest.RegionId = client.RegionId
+	describeScalingGroupRequest.ScalingGroupId1 = groupId
+
+	describeScalingGroupResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		return essClient.DescribeScalingGroups(describeScalingGroupRequest)
+	})
+
+	if err != nil {
+		return 0, 0, WrapErrorf(err, "failed to invoke DescribeScalingGroups,because of %v", err)
+	}
+	addDebug("DescribeScalingGroups", describeScalingGroupResponse, describeScalingGroupRequest)
+
+	resp, ok := describeScalingGroupResponse.(*ess.DescribeScalingGroupsResponse)
+
+	if ok != true {
+		return 0, 0, WrapError(fmt.Errorf("failed to parse DescribeScalingGroupsResponse of scaling group %s", groupId))
+	}
+
+	if resp.ScalingGroups.ScalingGroup == nil || len(resp.ScalingGroups.ScalingGroup) == 0 {
+		return 0, 0, WrapError(fmt.Errorf("the scaling group %s you specific is not found", groupId))
+	}
+
+	scalingGroup := resp.ScalingGroups.ScalingGroup[0]
+
+	if &scalingGroup == nil {
+		return 0, 0, WrapError(fmt.Errorf("the scaling group %s you specific is not found", groupId))
+	}
+
+	return scalingGroup.MinSize, scalingGroup.MaxSize, nil
+}
+
+// prepare kubeconf of kubernetes clsuter
+func DownloadUserKubeConf(client *connectivity.AliyunClient, clusterId string) (string, error) {
+
+	describeClusterUserKubeconfigRequest := cs.CreateDescribeClusterUserKubeconfigRequest()
+	describeClusterUserKubeconfigResponse, err := client.WithOfficalCSClient(func(csClient *cs.Client) (interface{}, error) {
+		describeClusterUserKubeconfigRequest.RegionId = client.RegionId
+		describeClusterUserKubeconfigRequest.ClusterId = clusterId
+		return csClient.DescribeClusterUserKubeconfig(describeClusterUserKubeconfigRequest)
+	})
+
+	if err != nil {
+		return "", WrapErrorf(err, "failed to describe user kuberconf of cluster %s,because of %v", clusterId, err)
```

Comment:
Please change line 363 to `return “”, WrapErrorf(err, DefaultErrorMsg, clusterId, describeClusterUserKubeconfigRequest.GetActionName(), AlibabaCloudSdkGoERROR) )`